Added hw10 task Figure

MariaSukhetska opened the pull request.

MariaSukhetska committed:
"added hw10 task Figure"

PakkiAlexey commented on src/ua/zhdanova/hw10/Figure.java:
```diff
@@ -0,0 +1,12 @@
+
+public class Figure {
+
+    public double perimeter() {
+        return 0.0;
+    }
+
+    public double area() {
+        return 0.0;
+    }
```
Please, make this methods abstract

PakkiAlexey commented on src/ua/zhdanova/hw10/Triangle.java:
```diff
@@ -0,0 +1,42 @@
+        return Math.sqrt(s * (s - sideA) * (s - sideB) * (s - sideC));
+    }
+
+    public double areaWithSinFormula() {
```
area methods should be overloading

PakkiAlexey commented on src/ua/zhdanova/hw10/Main.java:
```diff
@@ -0,0 +1,37 @@
+    }
+
+    private static void findFigureWithBiggestArea(Circle circle, Square square, Triangle triangle) {
+        if (circle.area() > square.area() && circle.area() > triangle.area()) {
```
Please, use array as a parameter

PakkiAlexey commented on src/ua/zhdanova/hw10/Main.java:
```diff
@@ -0,0 +1,37 @@
+
+    private static void findFigureWithBiggestArea(Circle circle, Square square, Triangle triangle) {
+        if (circle.area() > square.area() && circle.area() > triangle.area()) {
+            System.out.println("Area for circle is the biggest (" + circle.area() + ")");
+        } else if (square.area() > triangle.area()) {
+            System.out.println("Area for square is the biggest (" + square.area() + ")");
+        } else {
+            System.out.println("Area for triangle is the biggest (" + triangle.area() + ")");
```
You should use loop for the finding max area

MariaSukhetska committed:
"make abstract methods in Figure hw10"
"changed methods to overloading in Triangle hw10"
"used loop for the finding max area in Main hw10"

PakkiAlexey commented on src/ua/zhdanova/hw10/Triangle.java:
```diff
@@ -0,0 +1,41 @@
+        return 0.5 * sideA * sideB * Math.sin(angleInRadians);
+    }
+
+    public double areaWithBaseHeight() {
```
area methods should be overloading

MariaSukhetska committed:
"Updated hw10"
"Changed area methods to overloading"

PakkiAlexey commented on src/ua/zhdanova/hw10/Triangle.java:
```diff
@@ -0,0 +1,42 @@
+        return Math.sqrt(s * (s - sideA) * (s - sideB) * (s - sideC));
+    }
+
+    public double area(double sideA, double sideB, double angleInDegrees) {
+        double angleInRadians = Math.toRadians(angleInDegrees);
+        return 0.5 * sideA * sideB * Math.sin(angleInRadians);
+    }
+
+    public double area(double base, double height) {
+        return 0.5 * base * height;
+    }
+}
+
```
Please, use sides from the class state

MariaSukhetska committed:
"Update hw10"
"Use sides from the class state"

MariaSukhetska committed:
"Update hw10"
"Method area overloaded"

PakkiAlexey commented on src/ua/zhdanova/hw10/Triangle.java:
```diff
@@ -0,0 +1,42 @@
+        return Math.sqrt(s * (s - this.sideA) * (s - this.sideB) * (s - this.sideC));
+    }
+
+    public double area(double sideA, double sideB, double angleInDegrees) {
+        double angleInRadians = Math.toRadians(angleInDegrees);
+        return 0.5 * this.sideA * this.sideB * Math.sin(angleInRadians);
+    }
+
+    public static double area(double base, double height) {
+        return 0.5 * base * height;
+    }
```
Please, use sides from the  state

MariaSukhetska committed:
"Update hw10"
"use sides from the state"

MariaSukhetska committed:
"Update hw10"
"add method toString()"
